pkg/middleware/ratelimiter/map_limit_store.go

```diff
@@ -0,0 +1,87 @@
+package ratelimiter
+
+import (
+	"fmt"
+	"sync"
+	"time"
+)
+
+// MapLimitStore represents a data structure for in-memory storage of ratelimiter information
+type MapLimitStore struct {
+	// data is a map of key to limitValue
+	data map[string]limitValue
+	// mutex is a mutex for data map
+	mutex sync.RWMutex
+	// expirationTime is the time after which the data is considered expired
+	expirationTime time.Duration
+}
+
+// limitValue represents value of the limit counter
+type limitValue struct {
+	val        int64
+	lastUpdate time.Time
+}
+
+// NewMapLimitStore creates new in-memory data store for internal limiter data
+func NewMapLimitStore(expirationTime time.Duration, flushInterval time.Duration) (m *MapLimitStore) {
+	m = &MapLimitStore{
+		data:           make(map[string]limitValue),
+		expirationTime: expirationTime,
+	}
```

Comment:
This should return the interface type, that will ensure the type fully implements the interface:
```suggestion
// NewMapLimitStore creates new in-memory data store for internal limiter data
func NewMapLimitStore(expirationTime time.Duration, flushInterval time.Duration) LimitStore {
	m := &MapLimitStore{
		data:           make(map[string]limitValue),
		expirationTime: expirationTime,
	}
```

Reply:
`MapLimitStore` is an option in-memory implementation where others could be used; returning LimitStore doesn't make sense in this instance